### PR TITLE
Implement PXB-3091 - Remove Server Version Checks

### PR DESCRIPTION
**Pull request by altmannmarcelo**
https://jira.percona.com/browse/PXB-3091

Removed server version check. Going forward, PXB 8.0.X will support all versions from the same minor version (8.0).
Backing up any major versions will be blocked.

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
https://pxb.cd.percona.com/job/percona-xtrabackup-8.0-multijob/290/

All test failures are known.

File: storage/innobase/xtrabackup/src/xtrabackup.cc [hunk 1]
```diff
@@ -1128,7 +1128,7 @@ struct my_option xb_client_options[] = {
 
     {"no-server-version-check", OPT_NO_SERVER_VERSION_CHECK,
      "This option allows backup to proceed when the server version is greater "
-     "(newer) than the PXB supported version",
+     "(newer) than the PXB supported version. (This option is decprecated).",
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xtrabackup.cc]: deprecated

File: storage/innobase/xtrabackup/src/backup_mysql.cc [hunk 2]
```diff
@@ -370,7 +368,7 @@ static bool check_server_version(unsigned long version_number,
   }
 
   version_supported =
-      version_supported || (version_number > 80000 && version_number < 90000);
+      version_supported || (version_number > 80000 && version_number <= 80099);
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/backup_mysql.cc:371]: what about blocking higher versions like 8.1+ and 9.X and 10.X and so on? Separate PR?

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/backup_mysql.cc:371]: This is what this check does

80099 = [8].[00].[99] - Essentially we allow up to version 8.0.99.
8.1.0 is 80100

Comment (satya-bodapati) [reply storage/innobase/xtrabackup/src/backup_mysql.cc:371]: Ah, right. We don't need another major version check.

File: storage/innobase/xtrabackup/src/xtrabackup.cc (hunk 1 above)
Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/xtrabackup.cc]: fixed and force-pushed

**Review by satya-bodapati — APPROVED** [review]

**Pull request merged**